core/project_settings.h

```diff
@@ -218,4 +232,20 @@ Variant _GLOBAL_DEF_ALIAS(const String &p_var, const String &p_old_name, const V
 #define GLOBAL_DEF_ALIAS_RST(m_var, m_old_name, m_value) _GLOBAL_DEF(m_var, m_old_name, m_value, true)
 #define GLOBAL_GET(m_var) ProjectSettings::get_singleton()->get(m_var)
 
+/////////////////////////////////////////////////////////////////////////////////////////
+// Cached versions of GLOBAL_GET.
+// Cached but uses a typed variable for storage, this can be more efficient.
+#define GLOBAL_GET_CACHED(m_type, m_setting_name) ([](const char *p_name) -> const m_type & {                                               \
```

Comment:
```suggestion
#define GLOBAL_GET_CACHED(m_type, m_setting_name) ([](const char *p_name) -> const m_type {                                               \
```

I'm not sure it's safe to return a reference here (instead of a copy). Honestly, I'm surprised it compiles at all.